src/app/services/student-book-services/student-book.service.ts

```diff
@@ -95,12 +96,16 @@ export class StudentBookService {
     return window.URL.createObjectURL(blob);
   }
 
-  private addFilesToScheduleArray(data: Diary[]) {
+  private addFilesToScheduleArray(data: Diary[]): Diary[] {
     const scheduleWithFiles = data.map(el => {
       if (el.homeworkFileId) {
         this.getFileById(el.lessonId).subscribe(response => {
           el.blobUrl = this.b64toBlob(response.fileData, response.fileType);
           el.fileName = response.fileName;
+        },
+        err => {
+          console.error(err);
```

Comment:
why dont you show toast?

Reply:
It's not a mistake if there is no files.